mock-tee-primitives/src/identity.rs

```diff
@@ -67,97 +81,128 @@ pub enum Web2Network {
 
 #[derive(Encode, Decode, Clone, Debug, PartialEq, Eq, TypeInfo, MaxEncodedLen)]
 #[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
-pub enum IdentityWebType {
-	Web2(Web2Network),
-	Web3(Web3Network),
+pub struct SubstrateIdentity {
+	pub network: SubstrateNetwork,
+	pub address: Address32,
+}
+
+#[derive(Encode, Decode, Clone, Debug, PartialEq, Eq, TypeInfo, MaxEncodedLen)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub struct EvmIdentity {
+	pub network: EvmNetwork,
+	pub address: Address20,
 }
 
 #[derive(Encode, Decode, Clone, Debug, PartialEq, Eq, TypeInfo, MaxEncodedLen)]
 #[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
-pub enum IdentityHandle {
-	Address32([u8; 32]),
-	/// Its a 20 byte representation.
-	Address20([u8; 20]),
-	String(IdentityString),
+pub struct Web2Identity {
+	pub network: Web2Network,
+	pub address: IdentityString,
 }
 
 #[derive(Encode, Decode, Clone, Debug, PartialEq, Eq, TypeInfo, MaxEncodedLen)]
 #[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
-pub struct Identity {
-	pub web_type: IdentityWebType,
-	pub handle: IdentityHandle,
+pub enum Identity {
+	Substrate(SubstrateIdentity),
+	Evm(EvmIdentity),
+	Web2(Web2Identity),
```

Comment:
I'm basically fine with it, I'm just thinking if we should simplify it even more to flatten out the `SubstrateIdentity`/`EvmIdentity`/`Web2Identity`, so something like this:
```
	Substrate(SubstrateNetwork, Address32),
	Evm(EvmNetwork, Address20),
	Web2(Web2Network, IdentityString),
```

I'm tagging @zhizming-zhong @Verin1005 and @jonalvarezz as this change affects F/E too.

Reply:
I prefer the flatter one

Reply:
agree, the flatter version is more simpler

Reply:
Thank you, Kai.

What matters the most to clients is how easy it is to manipulate and traverse the objects.

Do I understand correctly that the following will be the corresponding Polkadot.js API types?

For `Substrate(SubstrateNetwork, Address32)`
```js
Identity: {
  Substrate: [SubstrateNetworkObject, '0x1234....']
}
```

For `Substrate(SubstrateNetworkIdentity)`
```js
Identity: {
  Substrate: {
    network: '...',
    address: '0x1234....'
  }
}
```

If so, we would prefer `Identity.Substrate.network` over `Identity.Substrate[1]`. Object properties over array indexes, in general.

Reply:
I actually think they should have the same polkadot-js types (i.e. without array indexes), but please correct me if I'm wrong @Verin1005 @zhizming-zhong

Reply:
Yes, they have the same value after encoding

Reply:
If that's the case, then there is not much difference from our side :)

The plain version looks easier to follow, though 👍